### ISSUE TITLE
In ResNet50, It is unequal to actual structure of library in the keras.applications.Resnet50

### ISSUE THREAD

**Issue by ahyunsoo2** [tensorflow/tensorflow#51238]
As you know, in the CNN, only layers of Convolution, BatchNormalization and Dense have weights. And Usually, they are constructed by this way. Input - Conv - BN - ReLU - Conv - BN - ReLU ... - Dense. But, As you can see, below the structure remain unusual.

![image](https://user-images.githubusercontent.com/83423286/128308372-357e542a-7197-46cc-9fce-536c7331bdce.png)

```
conv2_block1_0_conv/kernel:0
conv2_block1_0_conv/bias:0
conv2_block1_3_conv/kernel:0
conv2_block1_3_conv/bias:0
conv2_block1_1_bn/gamma:0
conv2_block1_1_bn/beta:0
conv2_block1_1_bn/moving_mean:0
conv2_block1_1_bn/moving_variance:0
conv2_block1_3_bn/gamma:0
conv2_block1_3_bn/beta:0
conv2_block1_3_bn/moving_mean:0
conv2_block1_3_bn/moving_variance:0
```
You can find this result by:

```
model = tf.keras.application.ResNet50()
#The unusual phenomenon begins with index 18.
model.weights[18]
```
_I recommend that you use debugging mode in your IDE. Then you'll find it easier._

In the below lines, the ResNet50 has stack_fn function for creating layers

```
def ResNet50():
.
.
  def stack_fn(x):
    x = stack1(x, 64, 3, stride1=1, name='conv2')
    x = stack1(x, 128, 4, name='conv3')
    x = stack1(x, 256, 6, name='conv4')
    return stack1(x, 512, 3, name='conv5')
.
.

```
In the below codes, the stack1 is for simplifying repeated residential blocks.

```
def stack1(x, filters, blocks, stride1=2, name=None):


  x = block1(x, filters, stride=stride1, name=name + '_block1')
  for i in range(2, blocks + 1):
    x = block1(x, filters, conv_shortcut=False, name=name + '_block' + str(i))
  return x
```

In the below structure, the block1 is Residential layers in ResNet50.

```
def block1(x, filters, kernel_size=3, stride=1, conv_shortcut=True, name=None):

  bn_axis = 3 if backend.image_data_format() == 'channels_last' else 1

  if conv_shortcut:
    shortcut = layers.Conv2D(
        4 * filters, 1, strides=stride, name=name + '_0_conv')(x)
    shortcut = layers.BatchNormalization(
        axis=bn_axis, epsilon=1.001e-5, name=name + '_0_bn')(shortcut) 
  else:
    shortcut = x

  x = layers.Conv2D(filters, 1, strides=stride, name=name + '_1_conv')(x)
  x = layers.BatchNormalization(
      axis=bn_axis, epsilon=1.001e-5, name=name + '_1_bn')(x)
  x = layers.Activation('relu', name=name + '_1_relu')(x)

  x = layers.Conv2D(
      filters, kernel_size, padding='SAME', name=name + '_2_conv')(x)
  x = layers.BatchNormalization(
      axis=bn_axis, epsilon=1.001e-5, name=name + '_2_bn')(x)
  x = layers.Activation('relu', name=name + '_2_relu')(x)

  x = layers.Conv2D(4 * filters, 1, name=name + '_3_conv')(x)
  x = layers.BatchNormalization(
      axis=bn_axis, epsilon=1.001e-5, name=name + '_3_bn')(x)

  x = layers.Add(name=name + '_add')([shortcut, x]) 
  x = layers.Activation('relu', name=name + '_out')(x)
  return x
```
My problem is why are the model instance different from the actual structures?

**Comment by Saduf2019** [comment]
@ahyunsoo2 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more refer to:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999

**Comment by ahyunsoo2** [comment]
As I got this answer, and I leave [it](https://stackoverflow.com/questions/68629607/in-keras-the-resnet50-has-a-strange-pattern).


@Saduf2019 
Next, I will do so in this case about _keras_.

Thank you for guiding me.
